### PR TITLE
updates via "rectorphp/rector"

### PR DESCRIPTION
**Pull request by voku**
- "global" -> remove unused "use" statements
- "phpunit" -> fix "@covers" comments
- "phpunit" -> use Iterator for "@dataProvider" (if possible)
- "phpunit" -> replace "->will($this->returnValue()" with "->willReturn()"
- "phpunit" -> replace "->will($this->throwException()" with "->willThrowException()"
- "UseTest.php" -> add missing namespace
- "composer.json" -> use "autoload-dev"

-> most of the changes are done automatically by "https://github.com/rectorphp/rector"

---
This change is [Reviewable](https://reviewable.io/reviews/nikic/php-parser/573)

### CODE REVIEW

File: test/PhpParser/Builder/PropertyTest.php [hunk 1]
```diff
@@ -92,57 +91,56 @@ public function testDefaultValues($value, $expectedValueNode) {
...
-                    ),
-                ])
-            ],
-            [
-                new Scalar\MagicConst\Dir,
-                new Scalar\MagicConst\Dir
-            ]
+    public function provideTestDefaultValues(): \Iterator
+    {
+        yield [
```
Comment (nikic) [inline test/PhpParser/Builder/PropertyTest.php]: The way PHPUnit works, using generators for data providers unfortunately doesn't make much sense. They will end up materialized as arrays anyway.

File: test/PhpParser/CodeTestAbstract.php [hunk 2]
```diff
@@ -2,11 +2,9 @@
 
 namespace PhpParser;
 
-use PHPUnit\Framework\TestCase;
-
 require_once __DIR__ . '/CodeTestParser.php';
```
Comment (nikic) [inline test/PhpParser/CodeTestAbstract.php]: With the autoload-dev change, possibly these requires can be dropped?

**Comment by nikic** [comment]
Not a fan of the `yield` changes, but otherwise this looks good.

File: test/PhpParser/CodeTestAbstract.php (hunk 2 above)
Comment (voku) [reply test/PhpParser/CodeTestAbstract.php]: correct :) these "require_once" usages in the tests have now been removed

**Comment by nikic** [comment]
Thanks!

File: test/PhpParser/ParserTest.php [hunk 3]
```diff
@@ -6,9 +6,8 @@
 use PhpParser\Node\Scalar;
 use PhpParser\Node\Scalar\String_;
 use PhpParser\Node\Stmt;
-use PHPUnit\Framework\TestCase;
 
-abstract class ParserTest extends TestCase
+abstract class ParserTest extends \PHPUnit\Framework\TestCase
```
Comment (TomasVotruba) [inline test/PhpParser/ParserTest.php]: These extends should be impoted with coding standrd run, either `--with-style` or standalone.
Just to be consistent with rest of the imports.

File: composer.json [hunk 4]
```diff
@@ -29,6 +29,11 @@
             "PhpParser\\": "lib/PhpParser"
         }
     },
+    "autoload-dev": {
+        "psr-4": {
+            "PhpParser\\": "test/PhpParser/"
+        }
+    },
```
Comment (TomasVotruba) [inline composer.json]: :+1:

**Pull request merged**